### PR TITLE
Add logic to delete feature for diet goals

### LINKED ISSUE #47: Add logic for user to delete diet goals
Extension of #11. Logic is required for user to delete the diet goals.

### PR DESCRIPTION
**Pull request by yicheng-toh**
Fixes #47

### CODE REVIEW

File: src/main/java/athleticli/ui/CommandName.java [hunk 1]
```diff
@@ -19,6 +19,7 @@ public class CommandName {
     public static final String COMMAND_DIET_GOAL_SET = "set-diet-goal";
     public static final String COMMAND_DIET_GOAL_EDIT = "edit-diet-goal";
     public static final String COMMAND_DIET_GOAL_LIST = "list-diet-goal";
+    public static final String COMMAND_DIET_GOAL_DELETE = "delete_diet-goal";
```
Comment (skylee03) [inline src/main/java/athleticli/ui/CommandName.java:22]:
```suggestion
    public static final String COMMAND_DIET_GOAL_DELETE = "delete-diet-goal";
```

File: src/main/java/athleticli/ui/Parser.java [hunk 2]
```diff
@@ -417,13 +421,30 @@ public static ArrayList<DietGoal> parseDietGoalSetEdit(String commandArgs) throw
...
     }
 
+    /**
+     * @param deleteIndexString Index of the goal to be deleted in String format
+     * @return Index of the goal in integer format in users' perspective.
+     * @throws AthletiException Catch invalid characters and numbers.
+     */
+    public static int parseDietGoalDelete(String deleteIndexString) throws AthletiException {
+        try {
+            int deleteIndex = Integer.parseInt((deleteIndexString.trim()));
```
Comment (skylee03) [inline src/main/java/athleticli/ui/Parser.java:437]:
```suggestion
            int deleteIndex = Integer.parseInt(deleteIndexString.trim());
```

File: src/main/java/athleticli/commands/diet/DeleteDietGoalCommand.java [hunk 3]
```diff
@@ -29,13 +35,16 @@ public DeleteDietGoalCommand(int deleteIndex) {
      * @return The message which will be shown to the user.
      */
     public String[] execute(Data data) throws AthletiException {
+        logger.log(Level.FINE,"Executing delete command for diet goals");
```
Comment (skylee03) [inline src/main/java/athleticli/commands/diet/DeleteDietGoalCommand.java:38]:
```suggestion
        logger.log(Level.FINE, "Executing delete command for diet goals");
```

File: src/main/java/athleticli/commands/diet/DeleteDietGoalCommand.java [hunk 4]
```diff
@@ -29,13 +35,16 @@ public DeleteDietGoalCommand(int deleteIndex) {
...
         DietGoalList dietGoals = data.getDietGoals();
         if (dietGoals.isEmpty()) {
             throw new AthletiException(Message.MESSAGE_DIETGOAL_EMPTY_DIETGOALLIST);
         }
         try {
             DietGoal dietGoalRemoved = dietGoals.get(deleteIndex - 1);
             dietGoals.remove(deleteIndex - 1);
+            logger.log(Level.FINE,String.format("Diet goals %s has been successfully removed",
+                    dietGoalRemoved.getNutrients()));
```
Comment (skylee03) [inline src/main/java/athleticli/commands/diet/DeleteDietGoalCommand.java:47]:
```suggestion
            logger.log(Level.FINE, String.format("Diet goals %s has been successfully removed",
                    dietGoalRemoved.getNutrients()));
```

**Review by skylee03 — COMMENTED** [review]
LGTM! There should be spaces after commas.
Suggestion: Use `FileHandler` to save the logger to a file instead of the default `ConsoleHandler` which prints the logs to `stderr`.

**Comment by yicheng-toh** [comment]
We should try to standardise on how to use the logger and what each level means.

**Review by skylee03 — APPROVED** [review]
LGTM!

**Pull request merged**